refactor(templates/README): standardize

michael-ciniawsky opened the pull request:
☝️ 😛

michael-ciniawsky commented:
Please `webpack-defaults` release after this :)

sapegin commented on templates/README.template:
```diff
@@ -1,112 +1,216 @@
-          </br>
-          Name
-        </a>
+        <img width="150" height="150"
```
Now images are not linked.

michael-ciniawsky replied:
Yep wrapping the image is the better solution 👌, I revert it :)

d3viant0ne replied:
As a part of the JS Foundation, we have to clearly define to consumers who the maintainers are.

We should probably default this to the org-maintainers ( not the admin group ) and not the bot as the bot breaks the above rule.

For any upgrade where the repo has assigned maintainers, they can update accordingly.

michael-ciniawsky replied:
@d3viant0ne This was my first intention aswell, but I wasn't sure if everyone would agree with that, so I used to non-personal bot account, will update

michael-ciniawsky replied:
https://github.com/orgs/webpack-contrib/teams/org-maintainers correct?

sapegin commented on templates/README.template:
```diff
@@ -1,112 +1,216 @@
 <div align="center">
   <!-- replace with accurate logo e.g from https://worldvectorlogo.com/ -->
-  <img width="200" height="200" src="https://cdn.worldvectorlogo.com/logos/javascript.svg">
+  <img width="140" height="140" src="https://cdn.worldvectorlogo.com/logos/javascript.svg">
```
Why project icon is smaller than webpack icon? ;-) I’d make `hspace` the same, otherwise they won’t be centered correctly.

d3viant0ne replied:
@michael-ciniawsky - Just use this one

```
  <img width="185" height="185" src="https://cdn.worldvectorlogo.com/logos/nodejs-icon.svg">
  <a href="https://webpack.js.org/">
    <img width="200" height="200" src="https://webpack.js.org/assets/icon-square-big.svg">
  </a>

```

michael-ciniawsky replied:
Done :)

sapegin commented on templates/README.template:
```diff
@@ -1,112 +1,216 @@
   </a>
-  <h1>${package}</h1>
-  <p>${description}</p>
+  <h1>${name}</h1>
```
There’s no `name` variable. Have you tested it? You need to remove the current Readme to see how it works.

michael-ciniawsky replied:
😱 I thought this was just a dummy placeholder sry..., obviously nope 😛, kk

michael-ciniawsky commented on templates/README.template:
```diff
@@ -1,112 +1,216 @@
 <div align="center">
   <!-- replace with accurate logo e.g from https://worldvectorlogo.com/ -->
-  <img width="200" height="200" src="https://cdn.worldvectorlogo.com/logos/javascript.svg">
+  <img width="140" height="140" src="https://cdn.worldvectorlogo.com/logos/javascript.svg">
```
The placeholder 'JS Logo' is to **fat** imho 😛  

> I’d make hspace the same, otherwise they won’t be centered correctly

? 🙃

sapegin replied:
Now it will render like this:

`[img] [hspace][img][hspace]`

This would be better:

`[hspace][img][hspace] [hspace][img][hspace]`

michael-ciniawsky commented:
**lib/tasks/readme.js**
```js
.apply({
        title: pkg.get('name')
                .split('-')
                .filter((name) => name !== 'webpack' ? name : '')
                .forEach((name) => name.charAt(0).toUpperCase())
                .join(' '),
        package: pkg.get('name'),
        description: pkg.get('description') || 'DESCRIPTION',
      })
```

### `title`

`name-loader` => `Name Loader`
`name-webpack-plugin` => `Name Plugin`


?

sapegin commented:
I like the idea, but I’d not remove ”webpack‘ ;-) So it’s just `_.startCase`.

michael-ciniawsky commented:
```js
//  Helpers
const title = (x) => x.split('-')
  .filter((name) => name !== 'webpack' ? name : '')
  .map((name) => name.replace(name.charAt(0), name.charAt(0).toUpperCase()))
  .join(' ')

const name = (x) => x.replace(/(-loader|-webpack-plugin)$/, '')
  .split('-')
  .map((name) => name.replace(name.charAt(0), name.charAt(0).toUpperCase()))
  .join('')

// 'Test' :)

console.log(name('some-loader'))
console.log(title('some-loader'))

// => 'Some Loader'
// => 'Some'

console.log(name('some-webpack-plugin'))
console.log(title('some-webpack-plugin'))

// => 'Some Plugin'
// => 'Some'

console.log(title('some-awesome-webpack-plugin'))
console.log(name('some-awesome-webpack-plugin'))

// => 'Some Awesome Plugin'
// => 'SomeAwesome'
```

```js
{
   test: /\.ext$/,
   use: [ { loader: `${name.toLowerCase()}-loader`, options: {} } ]
}
```


```js
plugins: [
  new `${name}Plugin(options)`
]
```

```js
{
   test: /\.ext$/,
   use: [ { loader: 'some-loader', options: {} }]
}
```

```js
plugins: [
  new SomePlugin(options)
]
```

michael-ciniawsky commented:
`-webpack` could also stay of course, but I personally see `-loader` && `-webpack-plugin` as `${package}` appendix  😛

sapegin commented:
They make it easy to understand what it is. Babel is not the same as Babel Loader and HTML is not the same as HTML Webpack Plugin (and I have no idea what’s HTML Plugin ;–)

michael-ciniawsky commented:
Fair point :) I leave `webpack` @bebraw @d3viant0ne any objections on this [in general]?

sapegin commented:
You can automate it ;-)

michael-ciniawsky commented:
@sapegin done 😊

sapegin commented on src/tasks/readme.js:
```diff
@@ -4,14 +4,28 @@ const { json, template } = require('mrm-core');
-        description: pkg.get('description') || 'DESCRIPTION',
-      })
-      .save();
+    const getCase = name => name.replace(
```
Why don’t you like Lodash? 👻

michael-ciniawsky replied:
I don't use `lodash`, therefore I didn't know `_.startCase` existed 😛 I interpreted your earlier comment as 'create helper fn' for casing :D. I'll take a look at `_.startCase` later today :ghost:

sapegin commented:
Are we going to finish this?

michael-ciniawsky commented:
Yep I intend to, but I merge on github a while ago and then it's seems I made a mistake during a rebase, so maybe I need to reopen this :) We take a look into later 😛

d3viant0ne commented on templates/README.template:
```diff
@@ -22,91 +23,200 @@ npm install --save-dev ${package}
           </br>
-          Name
+          Michael Ciniawsky
         </a>
```
Should add Alexander here and something else of note, when we first started applying this general readme template ...

1.) Tobias specifically asked that there be no more than 5 listed, ever.
2.) Asked not to be listed under the maintainers section.

michael-ciniawsky replied:
> 1.) Tobias specifically asked that there be no more than 5 listed, ever.

With Alexander added we are full then :): Will do...

> 2.) Asked not to be listed under the maintainers section.

You mean Tobias here ?

d3viant0ne replied:
As a maintainer listed in the readme of any repo under contrib.
